File: 3_Java/src/main/java/Algorithm/sorting/SelectionSort.java
package sorting;

public class SelectionSort {

    public static void selectionSort(int[] array) {

        int size = array.length;

        for(int i=0; i < size-1; i++) {

            int index = i;

            for(int j = i+1; j<size; j++) {

                if(array[index] > array[j]) {

                    index = j;

                }

            }

            int smallerNumber = array[index];
            array[index] = array[i];
            array[i] = smallerNumber;

        }

    }

}

File: 4_Javascript/algorithm/sorting/selection-sort.js
function selectionSort(array) {
    let size = array.length;
    for(let i=0; i<size-1; i++) {
        let index = i;
        for(let j=i+1; j<size; j++) {
            if(array[index] > array[j]){
                index = j;
            }
        }
        let smallerNumber = array[index];
        array[index] = array[i];
        array[i] = smallerNumber;
    }
}
